### PR TITLE
Do not hardcode the UDP listen port in ChipDeviceController

### PR DESCRIPTION
**Pull request by sagar-apple**
#### Problem

TLDR; Same PR as this https://github.com/project-chip/connectedhomeip/pull/9309, but for the C++ library 

ChipDeviceController hardcodes the UDP listen port to CHIP_PORT. This makes it impossible to have more than one instance of any of the controllers that use this on the same host. 
The target device is unable to tell which one is which and messages arrive to both which results in communication failure.

#### Change overview
Do not hardcode the UDP listen port.

#### Testing
* If manually tested, what platforms controller and device platforms were manually tested, and how?
Tested pairing with two instances of the chip-tool cli with the same accessory.

### CODE REVIEW

File: src/controller/CHIPDeviceController.cpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ DeviceController::DeviceController()
     mExchangeMgr              = nullptr;
     mStorageDelegate          = nullptr;
     mPairedDevicesInitialized = false;
-    mListenPort               = CHIP_PORT;
+    mListenPort               = 0;
```
Comment (tcarmelveilleux) [inline src/controller/CHIPDeviceController.cpp:112]: What about setting the port, rather than letting it be random? It would definitely help test cases to allow setting externally, before starting the listen in Init. While the default value may be zero, a setter called before init, or a new param in ControllerInitParams is required to make this follow usual convention of "a server can choose its port if it desires.

Comment (sagar-apple) [reply src/controller/CHIPDeviceController.cpp:112]: I'll push an update to this PR that adds the ListenPort to the init params.

File: src/controller/CHIPDeviceController.h [hunk 2]
```diff
@@ -102,6 +102,10 @@ struct ControllerInitParams
     ByteSpan controllerRCAC;
 
     uint16_t controllerVendorId;
+
+    /* The port used by the controller to listen for and send messages.
```
Comment (tcarmelveilleux) [inline src/controller/CHIPDeviceController.h:106]:
```suggestion
    /* The port used for operational communication to listen for and send messages over UDP/TCP.
```

**Review by tcarmelveilleux — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst ?

**Review by saurabhst — APPROVED** [review]
